types/react/index.d.ts

```diff
@@ -8,23 +8,74 @@ import * as CSS from "csstype";
 import * as PropTypes from "prop-types";
 import { Interaction as SchedulerInteraction } from "scheduler/tracing";
 
+/**
+ * @alias {@link AnimationEvent}
```

Comment:
These should be internal types that don't leak into user errors.

Reply:
OK, should I remove these aliases?

I was thinking about a convention for marking types as 'not intended for consumption', I'm not sure if `@internal` makes sense?

`@internal` can be used to strip types so might not be appropriate.

Reply:
It's not internal, it's just not part of the `declare module` and therefore cannot be imported from userspace. And since these are type-aliases, I was under the impression TypeScript resolves these so you wouldn't see e.g. `Expected NativeAnimationEvent but got string` and instead see `Expected AnimationEvent but got string`.

The JSDoc should be for userspace not for documentation of internals. So either we remove the JSDOC from the types simple aliases that are not in `declare module` scope or we add a normal comment explaining that these aliases leak out. Because eventually we wanted to have `React.Element` instead of `React.ReactElement` which would also require aliasing `Element` to `NativeElement`. If that aliasing isn't free, we should know.

Reply:
At which point the JSDOC would add no value. The alias is trivial so we don't need to add a comment just repeating that

Reply:
Agree